src/fu-util-common.c

```diff
@@ -185,16 +185,21 @@ fu_util_is_interesting_device (FwupdDevice *dev)
 gchar *
 fu_util_get_user_cache_path (const gchar *fn)
 {
+	const gchar *root = g_get_user_cache_dir ();
 	g_autofree gchar *basename = g_path_get_basename (fn);
 	g_autofree gchar *cachedir_legacy = NULL;
 
+	/* if run from a systemd unit, use the cache directory set there */
+	if (g_getenv ("CACHE_DIRECTORY") != NULL)
+		root = g_getenv ("CACHE_DIRECTORY");
+
```

Comment:
This is needed because of `DynamicUser=true`.  It causes it to use `/var/cache/private/fwupd` in practice rather than `/.cache/fwupd`